compiler/rustc_smir/src/rustc_smir/mod.rs

```diff
@@ -478,6 +487,37 @@ impl<'tcx> Stable<'tcx> for mir::UnOp {
     }
 }
 
+impl<'tcx> Stable<'tcx> for mir::AggregateKind<'tcx> {
+    type T = stable_mir::mir::AggregateKind;
+    fn stable(&self, tables: &mut Tables<'tcx>) -> Self::T {
+        match self {
+            mir::AggregateKind::Array(ty) => {
+                stable_mir::mir::AggregateKind::Array(tables.intern_ty(*ty))
+            }
+            mir::AggregateKind::Tuple => stable_mir::mir::AggregateKind::Tuple,
+            mir::AggregateKind::Adt(def_id, _, generic_arg, _, _) => {
```

Comment:
I guess we do want the rest of the fields from `mir::AggregateKind::Adt` cc @oli-obk

Reply:
I couldn't figure out how to add `VariantIdx` and `UserTypeAnnotationIndex` to `smir` so I went without them 😅

Reply:
if you leave something out, you can always make it opaque, so the information is at least there, even if useless except for printing

Reply:
Thanks, will keep that in mind

Reply:
Added missing fields let me know if it needs more work.

Reply:
That's exactly what was needed :).